libraries/AP_Proximity/AP_Proximity_LD06.cpp

```diff
@@ -0,0 +1,183 @@
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * ArduPilot device driver for Inno-Maker LD06 LiDAR
+ *
+ * ALL INFORMATION REGARDING PROTOCOL WAS DERIVED FROM InnoMaker DATASHEET:
+ *
+ * http://wiki.inno-maker.com/display/HOMEPAGE/LD06?preview=/6949506/6949511/LDROBOT_LD06_Development%20manual_v1.0_en.pdf
+ *
+ * Author: Adithya Patil, Georgia Institute of Technology
+ * Based on the SLAMTEC RPLiDAR code written by Steven Josefs, IAV GmbH and CYGBOT D1 LiDAR code
+ *
+ */
+
+#include "AP_Proximity_LD06.h"
+
+#if HAL_PROXIMITY_ENABLED && AP_PROXIMITY_LD06_ENABLED
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+#include <ctype.h>
+#include <stdio.h>
+
+// Add Hardware Abstraction Layer Library
+extern const AP_HAL::HAL& hal;
+
+// Update the sensor readings
+void AP_Proximity_LD06::update(void)
+{
+
+    // Escape if no connection detected/supported while running
+    if (_uart == nullptr) {
+        return;
+    }
+
+    // Begin getting sensor readings
+    // Calls method that repeatedly reads through UART channel
+    get_readings();
+
+    // Check if the data is being received correctly and sets Proximity Status
+    if (_last_distance_received_ms == 0) {
+        set_status(AP_Proximity::Status::NoData);
+    } else {
+        set_status(AP_Proximity::Status::Good);
+    }
+}
+
+// Stores the maximum distance that the sensor supports
+float AP_Proximity_LD06::distance_max() const
+{
+    return 12.0f;  // LD06 has a maximum range of 12m
+}
+
+// Stores the minimum distance that the sensor supports
+float AP_Proximity_LD06::distance_min() const
+{
+    return 0.02f;  // LD06 has a minimum range of 2 cm (0.02 m)
+}
+
+// Called repeatedly to get the readings at the current instant
+void AP_Proximity_LD06::get_readings()
+{
+
+    if (_uart == nullptr) {
+        return;
+    }
+
+    // Store the UART input in an array
+    uint32_t nbytes = _uart->available();
+
+    // Loops through all bytes that were received
+    while (nbytes-- > 0) {
+
+        // Gets and logs the current byte being read
+        uint8_t c = _uart->read();
+
+        // Stores the byte in an array if the byte is a start byte or we have already read a start byte
+        if(c == LD_START_CHAR || _response_data) {
+
+            // Sets to true if a start byte has been read, default false otherwise
+            _response_data = true;
+
+            // Stores the next byte in an array
+            _response[_byte_count] = c;
+            _byte_count++;
+
+            // The response should have 47 bytes in total
+            if (_byte_count == sizeof(_response)) {
+                parse_response_data();
+                _temp_boundary.update_3D_boundary(boundary);
+                _byte_count = 0;
+                _response_data = false;
+
+                set_status(AP_Proximity::Status::Good);
+            }
+        } else {
+            gcs().send_text(MAV_SEVERITY_CRITICAL, "LD06 Invalid Packet: %d", c);
```

Comment:
Use this macro instead as it allows  using the correct version of this on AP_Perip targets

```suggestion
            GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "LD06 Invalid Packet: %d", c);
```